event-svc/src/event/validator/event.rs

```diff
@@ -0,0 +1,151 @@
+use std::sync::Arc;
+
+use ceramic_core::{Cid, EventId};
+use ceramic_event::unvalidated;
+use ipld_core::ipld::Ipld;
+use recon::ReconItem;
+
+use crate::{
+    event::{service::InvalidItem, DeliverableRequirement},
+    store::{EventInsertable, SqlitePool},
+    Result,
+};
+
+#[derive(Debug)]
+pub struct ValidatedEvents {
+    /// These events are valid
+    pub valid: Vec<ValidatedEvent>,
+    /// We don't have enough information to validate these events yet.
+    /// e.g. The init event is required and the node has not yet discovered it.
+    pub pending: Vec<UnvalidatedEvent>,
+    /// Events that failed validation
+    pub invalid: Vec<InvalidItem>,
+}
+
+#[derive(Debug)]
+pub struct ValidatedEvent {
+    key: EventId,
+    cid: Cid,
+    event: Arc<unvalidated::Event<Ipld>>,
```

Comment:
yeah. might be worth a quick chat tomorrow? these types are just internal to the service until we have a "valid" or "pending" concept that is outside of this. It's really just to represent that it went through the flow and either was good, couldn't be verified yet and can either be pended or converted to an error, and was invalid. we could return our own type but I went with the public `InvalidItem` enum because we have better context about the "reason" at this point (which can be renamed per the other comment).

Reply:
I'm going to consider this resolved, but I can come back to it in the next (or a future PR)